atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapper.java

```diff
@@ -71,22 +75,21 @@ public String getShortTableNameFromMappingTable(
         }
     }
 
-    public Set<String> getLongTableNamesFromMappingTable(
-            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+    public Map<String, String> getShortToLongTableNamesFromMappingTable(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) {
         if (shortTableNames.isEmpty()) {
-            return Set.of();
+            return Map.of();
         }
-        ImmutableSet.Builder<String> longTableNames = ImmutableSet.builder();
+        ImmutableMap.Builder<String, String> shortToLongTableNames = ImmutableMap.builder();
         for (List<String> batch :
                 Iterables.partition(shortTableNames, AtlasDbConstants.MINIMUM_IN_CLAUSE_EXPRESSION_LIMIT)) {
-            longTableNames.addAll(getBatchOfLongTableNamesFromMappingTable(connectionSupplier, batch));
+            shortToLongTableNames.putAll(getBatchOfLongTableNamesFromMappingTable(connectionSupplier, batch));
         }
-        return longTableNames.build();
+        return shortToLongTableNames.buildOrThrow();
```

Comment:
Would we need to deal with the following case?

- I pass in 1,001 tables where the last element by iterator case-insensitively matches one of the earlier elements (so my set contains `aaa` and `Aaa` as element 1001 by iterator, let's say).
- Also, my set contains `bbb` where the table mapping is unavailable.
- So I load a batch of 1,000 and get 1,000 elements: if we had stored `aaa` and `Aaa` (and no `bbb`).
- And then I load the batch of just `Aaa` which stores the key-value pairs for `aaa` and `Aaa` once each
- and I throw, because that's how `buildOrThrow()` works

Reply:
oops, this is fine because you will throw the second time around: but you can imagine a suitable analogue with 1002 tables

Reply:
discussed offline: yeah, this case is messy enough that it's hard to say what we even _should_ do, so throwing and waiting for a manual response _is_ probably the best way to deal with it!